src/main/java/com/sap/oss/phosphor/fosstars/data/artifact/ReleaseInfoFromMaven.java

```diff
@@ -35,49 +36,86 @@ public class ReleaseInfoFromMaven extends AbstractReleaseInfoLoader {
    * Gathers release information about Maven artifacts.
    * 
    * @param mavenArtifact A {@link MavenArtifact}.
+   * @param start The start index to get the list of versions.
    * @return A {@link JsonNode} containing the Maven artifact release information.
    * @throws IOException If something goes wrong.
    */
-  private JsonNode mavenArtifactReleaseInfo(MavenArtifact mavenArtifact) throws IOException {
+  private JsonNode mavenArtifactReleaseInfo(MavenArtifact mavenArtifact, int start)
+      throws IOException {
     String groupQuery = URLEncoder.encode(String.format("g:\"%s\"", mavenArtifact.group()), UTF_8);
     String artifactQuery =
         URLEncoder.encode(String.format("a:\"%s\"", mavenArtifact.artifact()), UTF_8);
     String preparedQuery = String.format("%s+AND+%s", groupQuery, artifactQuery);
 
-    String url =
-        String.format("https://search.maven.org/solrsearch/select?q=%s&core=gav&wt=json&rows=10000",
-            preparedQuery);
+    String url = String.format(
+        "https://search.maven.org/solrsearch/select?q=%s&core=gav&wt=json&rows=100&start=%s",
+        preparedQuery, start);
     return fetch(url);
   }
 
-  @Override
-  public ReleaseInfoFromMaven update(Subject subject, ValueSet values) throws IOException {
-    Objects.requireNonNull(values, "Oh no! Values is null!");
-
-    MavenArtifact mavenArtifact = cast(subject, MavenArtifact.class);
-    JsonNode json = mavenArtifactReleaseInfo(mavenArtifact);
+  /**
+   * Gathers versions from Maven artifacts.
+   * 
+   * @param mavenArtifact A {@link MavenArtifact}.
+   * @param artifactVersions The set of {@link ArtifactVersion}s.
+   * @param start The start index to get the list of versions.
+   * @return A filled set {@link ArtifactVersion}s.
+   * @throws IOException If something goes wrong.
+   */
+  private Set<ArtifactVersion> fetchVersions(MavenArtifact mavenArtifact,
+      Set<ArtifactVersion> artifactVersions, int start) throws IOException {
+    JsonNode json = mavenArtifactReleaseInfo(mavenArtifact, start);
     JsonNode docs = json.at("/response/docs");
 
-    values.update(RELEASED_ARTIFACT_VERSIONS.unknown());
-    values.update(ARTIFACT_VERSION.unknown());
-
     if (docs.isArray()) {
-      Set<ArtifactVersion> artifactVersions = new HashSet<>();
       for (JsonNode item : docs) {
         if (item.has("v") && item.has("timestamp")) {
           artifactVersions.add(new ArtifactVersion(item.get("v").asText(),
               convertEpochToLocalDate(item.get("timestamp").asLong())));
         }
       }
+    }
+
+    int numFound = json.at("/response/numFound").asInt();
```

Comment:
null check on json.at("/response/numFound") would be safer here.

Reply:
It would not be necessary, as by default asInt() method will send a `0`

![image](https://user-images.githubusercontent.com/10615448/173385211-9c43111f-d98f-42e9-a2ad-42e75cdad61c.png)

Reply:
No you are calling asInt on  json.at("/response/numFound"),  json.at("/response/numFound") can be null

Reply:
I have already put in the test case to cover this https://github.com/SAP/fosstars-rating-core/pull/837/files#diff-b347ea38780d8fc3d0ffba6abf5d8f366385933eb84c5b466f84a31c0003fba3R137

Reply:
No its not about the test case. Will it not throw NPE if json.at("/response/numFound") is null?

Reply:
It doesn't throw NPE, if `numFound` is `null` or Not found. The test case confirms that.